node/src/components/contract_runtime/operations.rs

```diff
@@ -100,36 +100,40 @@ pub fn execute_finalized_block(
         state_root_hash = state_hash;
     }
 
-    let execution_results_root_hash = compute_execution_results_root_hash(
-        &mut execution_results.iter().map(|(_, _, result)| result),
-    )?;
+    // Write the deploy approvals and execution results Merkle root hashes to global state if there
+    // were any deploys.
     let block_height = finalized_block.height();
+    if let Some(deploy_approvals_root_hash) = maybe_deploy_approvals_root_hash {
+        let execution_results_root_hash = compute_execution_results_root_hash(
```

Comment:
Why not check for the the block effects as well?

Reply:
We only call `compute_execution_results_root_hash` if the set of deploys is non-empty.  I guess I could print an error in there if we pass an empty set of deploys, but seems low risk to me.